src/widget/wstarrating.cpp

```diff
@@ -37,22 +37,26 @@ QSize WStarRating::sizeHint() const {
     QStyleOption option;
     option.initFrom(this);
     QSize widgetSize = style()->sizeFromContents(QStyle::CT_PushButton, &option,
-                                                 m_starRating.sizeHint(), this);
+                                                 m_visualStarRating.sizeHint(), this);
 
     m_contentRect.setRect(
-        (widgetSize.width() - m_starRating.sizeHint().width()) / 2,
-        (widgetSize.height() - m_starRating.sizeHint().height()) / 2,
-        m_starRating.sizeHint().width(),
-        m_starRating.sizeHint().height()
+        (widgetSize.width() - m_visualStarRating.sizeHint().width()) / 2,
+        (widgetSize.height() - m_visualStarRating.sizeHint().height()) / 2,
+        m_visualStarRating.sizeHint().width(),
+        m_visualStarRating.sizeHint().height()
     );
 
     return widgetSize;
 }
 
-void WStarRating::slotSetRating(int rating) {
-    m_starRating.setStarCount(rating);
-    m_currRating = rating;
-    update();
+void WStarRating::slotSetRating(int starCount) {
+    if (starCount == m_starCount) {
+        // Unchanged
+        return;
+    }
+    m_starCount = starCount;
+    resetVisualRating();
+    emit ratingChanged(starCount);
```

Comment:
shouldn't this be `updateVisualrating()`?
Also, emitting the signal with the same value after we may just have received the new value doesn't make sense to me. What for?
It seems emit is required here only because you use `slotSetRating` instead of emit in `mouseReleaseEvent`.